Subject: Avoid messages when quiet
From: Robert Schwebel <r.schwebel@pengutronix.de>

I've tried to use busybox with CONFIG_FEATURE_EXTRA_QUIET, but init's
startup banner does still appear. I don't understand the intended logic
behind the sniplet patched below: it is in an 'else' part that is only
executed if the syslog feature is not enabled; why should something
printed out differently if syslog is not enabled?

So this patch removes the duplicate full_write and delegates it to the
normal one.

The second part of the patch changes the file descriptor to stdout
instead of stderr. Init doesn't print out errors.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>

ported to busybox-1.13.2

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>

ported to busybox-1.15.2

Signed-off-by: Ladislav Michl <ladis@linux-mips.org>

---
 init.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: busybox-1.13.2/init/init.c
===================================================================
--- busybox-1.13.2.orig/init/init.c
+++ busybox-1.13.2/init/init.c
@@ -148,17 +148,19 @@ static void message(int where, const cha
 				}
 			}
 		}
+#if 0
 		if (where & L_LOG) {
 			full_write(log_fd, msg, l);
 			if (log_fd == STDERR_FILENO)
 				return; /* don't print dup messages */
 		}
+#endif
 	}
 #endif
 
 	if (where & L_CONSOLE) {
 		/* Send console messages to console so people will see them. */
-		full_write(STDERR_FILENO, msg, l);
+		full_write(STDOUT_FILENO, msg, l);
 	}
 }
 
